lib/ansible/modules/cloud/amazon/aws_api_gateway.py

```diff
@@ -324,5 +361,22 @@ def create_deployment(client, api_id=None, stage=None, description=None):
     return client.create_deployment(restApiId=api_id, stageName=stage, description=description)
 
 
+@AWSRetry.backoff(**retry_params)
+def get_rest_apis(client, limit=500):
+    return client.get_rest_apis(limit=limit)
```

Comment:
Please use the paginators and return the full result set

http://boto3.readthedocs.io/en/latest/reference/services/apigateway.html#APIGateway.Paginator.GetRestApis

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#api-throttling-and-pagination